editoast/src/core/mq_client.rs

```diff
@@ -156,15 +354,22 @@ impl RabbitMQClient {
         published_payload: &Option<T>,
         mandatory: bool,
         override_timeout: Option<u64>,
-    ) -> Result<MQResponse, Error>
+    ) -> Result<MQResponse, MqClientError>
     where
         T: Serialize,
     {
-        // Create a channel
-        let connection = self.pool.get().await.map_err(Error::DeadpoolLapin)?;
-        let channel = connection.create_channel().await.map_err(Error::Lapin)?;
+        let correlation_id = Uuid::new_v4().to_string();
+
+        // Get the next channel
```

Comment:
Should we put from this line until the `drop(channel_work)` into a function (would avoid to manually drop it and it also makes sense as a block of computation? What do you think?

Reply:
I'd rather keep the explicit drop to mark the moment we give it back to the pool (this is really important and I do not want since to be on a side effect : keeping it explicit will keep a trace that this it required), wdyt ?